Add warning about agreement on Client certificate bytes. (#76)
tlswg/tls-exported-authenticator: draft-ietf-tls-exported-authenticator.md

"Exported Authenticator" and therefore cannot be transplanted into other
protocols.

In TLS 1.3 the client and server are not guaranteed to agree on the client’s

> final flight until the first application message.  Because EAs can be

Reviewer: As noted, this first is false.  0-RTT or 0.5-RTT both violate this.

Reviewer 2: MT is right. You need to delete the first sentence. just the second two are sufficient. Otherwise, this is fine.

Reviewer 3: We do get a guarantee of agreement eventually, though nailing down a concise way to describe exactly when that happens may take some thought.  IIRC it's successful use of application_traffic_secret_0.
I think the phrasing would be a little awkward if we just jump straight in as "Because EAs can be negotiated out-of-band it is possible to negotiate EAs without agreeing on the entire transcript".  If we don't want to mention the lack of transcript agreement at all, we might flip it around, "It is possible to negotiate EAs without agreeing on the entire transcript, as might occur if [...]".  We might also make the first sentence blander without removing it entirely, to mention that the non-agreement on the client's final flight is possible/a risk, without making a specific claim of when the risk goes away.

Reviewer 2: I don't believe that that's true. application_traffic_secret_0 does not include anything from the client's last flight.

Author: What do people think of this text instead:
```
In TLS 1.3 the client does not explicitly learn whether its Finished message
was accepted. Because the application traffic keys are not dependent on the
client's final flight, the client cannot learn whether the server ever received
it.  To avoid disagreement between the client and server on the authentication
status of EAs, servers MUST verify the client Finished before sending an EA.
```

Reviewer 3: That seems like a good approach.  I might spend a few words for "client cannot learn just from TLS messages" or similar, since some application protocols on top of TLS might enable the client to do so.  But I don't think it's strictly necessary to have such a clarification.

negotiated out-of-band it is possible to negotiate EAs without agreeing on the

> entire transcript.  Servers SHOULD send application data before sending an

Reviewer 4: ```suggestion
entire transcript.  Servers SHOULD send application data before sending a
```

> Finished message was accepted. Because the application traffic keys are not
> dependent on the client's final flight, the client cannot learn whether the
> server ever received it.  To avoid disagreement between the client and server

Reviewer 2: ```suggestion
Finished message was accepted. Because the application traffic keys are not
dependent on the client's final flight, receiving messages from the server
does not prove that the server received the client's Finished. To avoid disagreement between the client and server
```

CertificateRequest to the client.  If there is no application data to send the

> server MAY send a NewSessionTicket.

Reviewer 3: NewSessionTicket is not an application message, so the previous text does not actually say that receiving NST would suffice to produce full-transcript agreement.  (It would, of course, since NST is protected by application traffic keys, but we should probably tweak some text to be more precise.)

Reviewer 5: Suggest adding "(because that is also protected by application traffic keys)."
